services/task-svc/internal/task/task.go

```diff
@@ -60,11 +60,22 @@ func (s ServiceServer) CreateTask(ctx context.Context, req *pb.CreateTaskRequest
 		description = *req.Description
 	}
 
+	// When changing this array also adjust the error message below
+	allowedInitialStatuses := []pb.TaskStatus{pb.TaskStatus_TASK_STATUS_IN_PROGRESS, pb.TaskStatus_TASK_STATUS_IN_PROGRESS}
```

Comment:
Looks duplicated here. `TaskStatus.TASK_STATUS_TODO`?

```suggestion
	allowedInitialStatuses := []pb.TaskStatus{pb.TaskStatus.TASK_STATUS_TODO, pb.TaskStatus_TASK_STATUS_IN_PROGRESS}
```